.github/workflows/run-framework-tests.yaml

```diff
@@ -0,0 +1,53 @@
+on:
+  pull_request:
+    types: [opened, reopened]
```

Comment:
This should probably be run on "push". Do you wanna check the existing Flutter workflow triggers?

Reply:
The intent of this workflow  is to add checks to the PR before it's merged so you know if merging it would break the tests. I think we want it on [opened, reopened] for every PR. We should be able to add some settings to the repo to require these framework test checks to pass before it can be merged as well. Probably should add synchronized to the list for cases when PRs are updated.

Maybe I'm missing something, but I don't see something in flutter/flutter or flutter/engine for this in the .github/workflows, but they definitely do have the same style of checks that run on PRs.